src/main/java/org/bdgenomics/convert/ga4gh/BdgenomicsAlignmentRecordToGa4ghReadAlignment.java

```diff
@@ -0,0 +1,183 @@
+import org.bdgenomics.convert.AbstractConverter;
+import org.bdgenomics.convert.Converter;
+import org.bdgenomics.convert.ConversionException;
+import org.bdgenomics.convert.ConversionStringency;
+
+import org.bdgenomics.formats.avro.AlignmentRecord;
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.CigarOperator;
+import htsjdk.samtools.TextCigarCodec;
+import htsjdk.samtools.ValidationStringency;
+import htsjdk.samtools.CigarElement;
+
+
+
+import org.slf4j.Logger;
+
+import java.util.*;
+
+/**
+ * Convert bdg-formats AlignmentRecord to GA4GH ReadAlignment.
+ */
+@Immutable
+final class BdgenomicsAlignmentRecordToGa4ghReadAlignment extends AbstractConverter<AlignmentRecord, Reads.ReadAlignment> {
+
+    /**
+     * Convert bdg-format AlignmentRecord to GA4GH ReadAlignment.
+     */
+    BdgenomicsAlignmentRecordToGa4ghReadAlignment() { super(AlignmentRecord.class, Reads.ReadAlignment.class); }
+
+    private List<ga4gh.Reads.CigarUnit> convertCigar(String cigarString) {
+        if (cigarString == null) {
+            return Collections.emptyList();
+        }
+        else {
+            htsjdk.samtools.Cigar cigar = htsjdk.samtools.TextCigarCodec.decode(cigarString);
+
+            List result = new ArrayList<ga4gh.Reads.CigarUnit>();
+            for( CigarElement element : cigar.getCigarElements() ) {
+                Reads.CigarUnit.Builder cuBuilder = ga4gh.Reads.CigarUnit.newBuilder();
+                cuBuilder.setOperationLength(element.getLength());
+                if(element.getOperator() == CigarOperator.M)  cuBuilder.setOperation(Reads.CigarUnit.Operation.ALIGNMENT_MATCH);
+                if(element.getOperator() == CigarOperator.I)  cuBuilder.setOperation(Reads.CigarUnit.Operation.INSERT);
+                if(element.getOperator() == CigarOperator.D)  cuBuilder.setOperation(Reads.CigarUnit.Operation.DELETE);
+                if(element.getOperator() == CigarOperator.N)  cuBuilder.setOperation(Reads.CigarUnit.Operation.SKIP);
+                if(element.getOperator() == CigarOperator.S)  cuBuilder.setOperation(Reads.CigarUnit.Operation.CLIP_SOFT);
+                if(element.getOperator() == CigarOperator.H)  cuBuilder.setOperation(Reads.CigarUnit.Operation.CLIP_HARD);
+                if(element.getOperator() == CigarOperator.P)  cuBuilder.setOperation(Reads.CigarUnit.Operation.PAD);
+                if(element.getOperator() == CigarOperator.EQ)  cuBuilder.setOperation(Reads.CigarUnit.Operation.SEQUENCE_MATCH);
+                if(element.getOperator() == CigarOperator.X)  cuBuilder.setOperation(Reads.CigarUnit.Operation.SEQUENCE_MISMATCH);
+
+                result.add(cuBuilder.build());
+
+            }
+
+            return result;
+        }
+
+    }
+
+    @Override
+    public ga4gh.Reads.ReadAlignment convert(final AlignmentRecord alignmentRecord,
+                                       final ConversionStringency stringency,
+                                       final Logger logger) throws ClassCastException {
```

Comment:
this method should `throw ConversionException`, catch ClassCastException and wrap it